Week2/akif.js

```diff
@@ -0,0 +1,108 @@
+// Part 1 Hello World
+console.log('Hello World'); // English
+console.log('Merhaba Dunya'); // Turkish
+console.log('Hallo Wereld'); // Dutch
+console.log('مرحبا بالعالم'); // Arabic
+
+// Part 2 Error  -----------------------------------------
+console.log("I'm awesome");
+
+// Part 3 Console.log(x)   --------------------------------
+let x;
+console.log('the value of my variable x will be: numberOfTheStudents');
+console.log(x);
+x = 16;
+console.log('the value of my variable x will be: numberOfTheStudents');
+console.log(x);
+
+// Part 4  String y  ---------------------------------------
+let y = 'I love you JavaScript';
+console.log('the value of my variable y will be: feelingAboutJavaScript');
+console.log(y);
+y = 'I will learn you';
+console.log('the value of my variable y will be: myPlanAboutJavaScript');
+console.log(y);
+
+// Part 5  Round a number  ---------------------------------
+let z = 7.25;
+console.log(z);
+let a = Math.round(z);
+console.log(a);
+let b = Math.max(z, a);
+console.log(b);
+
+// Part 6 Arrays   ------------------------------------------
+let myNumbers = [];
+console.log('the value of my variable y will be: myLuckyNumbers');
+console.log(myNumbers);
+let myFavouriteAnimals = ['cat', 'butterfly', 'bear'];
+console.log(myFavouriteAnimals);
+myFavouriteAnimals.push('baby pig');
+console.log(myFavouriteAnimals);
+
+// Part 7  More Strings - Length -----------------------------
+let myString =
+  'JavaScript (JS) is a lightweight interpreted or JIT-compiled programming language with first-class functions.';
+console.log(myString);
+console.log(myString.length);
+
+// Part 8  typeof  -------------------------------------------
+let k = 42;
+let l = 'Hello World';
+let m;
+let n = true;
+console.log(k, l, m, n);
+console.log('the types of my variable accordingly are: number, string, undefined, boolean');
+console.log(typeof k, typeof l, typeof m, typeof n);
+if (typeof k != typeof l) {
+  console.log('Not Same Type');
+}
+if (typeof k != typeof m) {
+  console.log('Not Same Type');
+}
+if (typeof k != typeof n) {
+  console.log('Not Same Type');
+}
+if (typeof l != typeof m) {
+  console.log('Not Same Type');
+}
+if (typeof l != typeof n) {
+  console.log('Not Same Type');
+}
+if (typeof m != typeof n) {
+  console.log('Not Same Type');
+}
+
+// Part 9   Remainder %  -----------------------------------------
+x = 7;
+x %= 3;
+console.log(x);
+
+y = 100;
+console.log((y %= 15));
+
+if (y % 2 === 0) {
+  console.log(y + ' is a even number');
+}
+y = 13;
+if (y % 2 === 1) {
+  console.log(y + ' is a odd number');
+}
+
+// Part 10 --------------------------------------------------------------
+let arr = [13, 'I am a programmer', undefined, null];
+console.log(arr);
+
+function arrayTypeChecker(b) {
+  for (let i = 0; i < b.length; i++) {
+    if (typeof b[i] !== typeof b[i + 1]) {
```

Comment:
Hi Akif - although your logic of checking for types of different array elements is correct this way, it has got one small bug.

You are running the loop from 0 until `i` is less than array length which in our case is 4. That means, our last iteration would have `i = 3/b[i] = null` and then when execution reaches line 98, it would try to equate `typeof b[3]` with `typeof b[4]` which doesn't exist and also goes against what we wanted to actually do.

This can be fixed easily if you ran the loop until `i < b.length - 1`. This way, our last iteration would compare `typeof b[2]` and `typeof b[3]` and would then exit out from the loop.